src/transaction.rs

```diff
@@ -94,27 +144,47 @@ impl SigningProcess {
 
         println!();
         println!("1️⃣ Step 1a: Creating your own commitment.");
-        let cp = CommitmentPair::generate_default_csprng();
+
         let own_pk = wallet.public_key();
-        let commitment_str = hex::encode(cp.commitment().to_bytes());
+    
 
-        let secret = secret_to_vec(cp.random_secret());
-        let encrypted_secret = encrypt(&secret, password.as_ref())?;
+        // We should add multiple (e.g., 2) pre-commitments
+        let mut i = 0;
+        let mut encrypted_secrets = vec![];
+        let mut cp_list = vec![];
+        let mut commitment_str_list = vec![];
+        while i < MUSIG2_PARAMETER_V {
+            let cp = CommitmentPair::generate_default_csprng();
+            let commitment_str = hex::encode(cp.commitment().to_bytes());
 
-        println!("❗️ Please give to your co-signers the following information:");
-        println!("Public Key: {}", own_pk);
-        println!("Commitment: {}", commitment_str);
+            let secret = secret_to_vec(cp.random_secret());
+            let encrypted_secret = encrypt(&secret, password.as_ref())?;
 
-        let mut state = SigningProcess {
-            encrypted_secret,
+            encrypted_secrets.push(encrypted_secret);
+            cp_list.push(cp);
+            commitment_str_list.push(commitment_str);
+
+            i += 1;
+        }
+
+        let mut state = SigningProcessNew {
+            encrypted_secret_list: encrypted_secrets,
             own_public_key: own_pk,
-            own_commitment: cp,
-            other_commitments: vec![],
+            own_commitment_list: cp_list,
+            other_commitments_list: vec![],
             transaction: None,
             partial_signatures: vec![],
             filename,
         };
 
+        println!("❗️ Please give to your co-signers the following information:");
+        println!("Public Key: {}", own_pk);
+        // println!("Commitments: {}", );
+        for c in commitment_str_list.iter() {
+            println!("{}", c);
+        }
```

Comment:
Maybe make printout clearer:
```suggestion
        for (i, c) in commitment_str_list.iter().enumerate() {
            println!("Commitment {}: {}", i, c);
        }
```

Reply:
Actually, perhaps start counting at 1, so print i+1 instead

Reply:
Actually, perhaps start counting at 1, so print i+1 instead